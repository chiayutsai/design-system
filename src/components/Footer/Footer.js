/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright © 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import useStyles from 'isomorphic-style-loader/useStyles'
import { useDispatch } from 'react-redux'
import { setModal } from 'actions/modal'
import { PRIVACY } from 'constants/modal'
import Link from '../Link'
import styles from './Footer.scss'

const Footer = () => {
  useStyles(styles)

  const dispatch = useDispatch()
  const onPrivacyClick = () => dispatch(setModal({ name: PRIVACY }))

  return (
    <div className={styles.root}>
      <div className={styles.container}>
        <span className={styles.text}>© Your Company</span>
        <span className={styles.spacer}>·</span>
        <Link className={styles.link} to="/">
          Home
        </Link>
        <span className={styles.spacer}>·</span>
        <Link className={styles.link} to="/admin">
          Admin
        </Link>
        <span className={styles.spacer}>·</span>
        <span
          role="presentation"
          className={styles.link}
          onClick={onPrivacyClick}>
          Privacy
        </span>
        <span className={styles.spacer}>·</span>
        <Link className={styles.link} to="/not-found">
          Not Found
        </Link>
      </div>
    </div>
  )
}

export default Footer
