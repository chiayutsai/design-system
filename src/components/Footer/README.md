This component is just for DEMO, it can be deleted later.
