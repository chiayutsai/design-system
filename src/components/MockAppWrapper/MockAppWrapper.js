import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
