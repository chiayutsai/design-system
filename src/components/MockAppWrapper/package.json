{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
