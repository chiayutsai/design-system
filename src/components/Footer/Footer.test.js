import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import Footer from './Footer'

describe('Footer', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Footer />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
