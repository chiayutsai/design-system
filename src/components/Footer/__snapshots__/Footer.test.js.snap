// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render correctly 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <span
      className="text"
    >
      © Your Company
    </span>
    <span
      className="spacer"
    >
      ·
    </span>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <span
      className="spacer"
    >
      ·
    </span>
    <a
      className="link"
      href="/admin"
      onClick={[Function]}
    >
      Admin
    </a>
    <span
      className="spacer"
    >
      ·
    </span>
    <span
      className="link"
      onClick={[Function]}
      role="presentation"
    >
      Privacy
    </span>
    <span
      className="spacer"
    >
      ·
    </span>
    <a
      className="link"
      href="/not-found"
      onClick={[Function]}
    >
      Not Found
    </a>
  </div>
</div>
`;
